service/src/main/java/org/cbioportal/service/impl/MutationCountServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2016 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+ /*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.cbioportal.service.impl;
+
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import org.cbioportal.model.PositionMutationCount;
+import org.cbioportal.persistence.MutationCountRepository;
+import org.cbioportal.service.MutationCountService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ *
+ * @author abeshoua
+ */
+@Service
+public class MutationCountServiceImpl implements MutationCountService {
+	
+	@Autowired
+	private MutationCountRepository mutationCountRepository;
+	
+	@Override
+	public List<PositionMutationCount> getPositionMutationCounts(List<String> genes, List<List<Integer>> positions) {
+		List<PositionMutationCount> counts = new LinkedList<>();
+		Iterator<String> geneIt = genes.iterator();
+		Iterator<List<Integer>> positionIt = positions.iterator();
+		while (geneIt.hasNext()) {
+			if (!positionIt.hasNext()) {
```

Comment:
I see .. if we don't have the same count in the gene list and the list of position lists, then we are going to stop processing excess genes. But maybe it would be best to do a simple test and if size(gene array) != size(positions list array) then the call should return with an error status (400 Bad Request) and message.

If the gene array is too large, we probably should not be ignoring the "surplus" genes .. or if the gene list is too short we probably should not assume that the surplus position lists which were passed in are meaningless.

Or maybe I am confused here. I would expect that for every gene, we would get a position list (perhaps an empty one) .. so that they have a 1 to 1 match